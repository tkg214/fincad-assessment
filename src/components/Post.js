import React from 'react';
import PropTypes from 'prop-types';

const Post = props => {
  return (
    <div>
      <h5 data-test="name">{`by ${props.name}`}</h5>
      <p data-test="body">
        {props.body}
      </p>
    </div>
  );
};

Post.propTypes = {
  name: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired
};

export default Post;
