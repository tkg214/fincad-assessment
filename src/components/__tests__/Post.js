import React from 'react';
import { render, mount } from 'enzyme';

import Post from '../Post';

test('the comment component renders with defaults', () => {
  const wrapper = renderPost();
  expect(wrapper).toMatchSnapshot();
});

test('the component renders with props passed in', () => {
  const name = 'Test Name';
  const body = 'Test Body';
  const wrapper = mountPost({ name, body });
  expect(wrapper.find(sel('name')).text()).toBe(`by ${name}`);
  expect(wrapper.find(sel('body')).text()).toBe(body);
});

function getPostProps(props = {}) {
  return {
    name: '',
    body: '',
    ...props
  };
}

function renderPost() {
  const postProps = getPostProps();
  return render(<Post {...postProps} />);
}

function mountPost(props) {
  const postProps = getPostProps(props);
  return mount(<Post {...postProps} />);
}

function sel(id) {
  return `[data-test="${id}"]`;
}
