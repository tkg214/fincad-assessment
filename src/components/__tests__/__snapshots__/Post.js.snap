// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the comment component renders with defaults 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {
              "data-test": "name",
            },
            "children": Array [
              Object {
                "data": "by ",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h5",
            "next": Object {
              "attribs": Object {
                "data-test": "body",
              },
              "children": Array [],
              "name": "p",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
          },
          Object {
            "attribs": Object {
              "data-test": "body",
            },
            "children": Array [],
            "name": "p",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "data-test": "name",
              },
              "children": Array [
                Object {
                  "data": "by ",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "h5",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
            },
            "type": "tag",
          },
        ],
        "name": "div",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
